src/storage-types/src/controller.rs

```diff
@@ -48,6 +48,24 @@ pub struct CollectionMetadata {
     /// The shard id of the persist-txn shard, if `self.data_shard` is managed
     /// by the persist-txn system, or None if it's not.
     pub txns_shard: Option<ShardId>,
+
+    /// The ID of the instance on which the collection runs.
+    ///
+    /// When creating collections, this will be `None` and calls to
+    /// `create_collections` will populate it.
```

Comment:
isn't `create_collections` the thing creating this struct anyways?

Reply:
Yes but it's in a different module